metadata-ingestion/src/datahub/ingestion/source/state/ldap_state.py

```diff
@@ -0,0 +1,48 @@
+from typing import Iterable, List
+
+import pydantic
+
+from datahub.ingestion.source.state.stale_entity_removal_handler import (
+    StaleEntityCheckpointStateBase,
+)
+
+
+class LdapCheckpointState(StaleEntityCheckpointStateBase["LdapCheckpointState"]):
+    """
+    Base class for representing the checkpoint state for all LDAP based sources.
+    Stores all ldap_users being ingested and is used to remove any stale entities.
+    """
+
+    encoded_ldap_users: List[str] = pydantic.Field(default_factory=list)
+
+    @classmethod
+    def get_supported_types(cls) -> List[str]:
+        return ["corpuser"]
```

Comment:
For completeness, we should also support `corpGroup`

Reply:
If it's easier to implement, you can take a look at the Tableau source for an example of how to make this CheckpointState much simpler

Reply:
This has been completed